noria-server/src/controller/inner.rs

```diff
@@ -1117,10 +1124,11 @@ impl ControllerInner {
             .count()
             > 0
         {
-            // This query leaf node has children -- typically, these are readers, but they can
-            // also be egress nodes or other, dependent queries. If the reader has replicas, we
-            // are looking for a single egress node that connects the leaf to replicas in other
-            // domains.
+            // This query leaf node has children -- egress nodes or other, dependent queries.
+            // Typically, we are looking for a single egress node that connects the leaf to
+            // readers in other domains.
+            // TODO: if each reader (including the 1st) definitely gets its own domain, this
+            // code can be a lot simpler.
```

Comment:
How much is "a lot"? I think we should favor simplicity then.

Reply:
👍